### PR TITLE
Dance bomb

### PR DESCRIPTION
**Pull request by Alanine42**
TLDR: press WASD together, grab the egg and throw

----

Activation: client 0 presses WASD together -> egg turns into the dance bomb
- change egg tag, shines purple light

Pickup
- by collision with player
- starts a 5-second detonation countdown (when you hold it). 

Throwing (press E)
- starts a 1-second detonation countdown. 

Colliding with player or ground:
- fast-decrease the detonation countdown to 5 or 8 ticks (or keep decreasing the current countdown, whichever is smaller)
- the countdown is shown in UI (bottom left)

Explosion
- effect radius = 5. 
- gain 10 points from each player that you made dancing 

Dance in action:
- four twinkling points lights 
- I made the shader do all the heavy lifting, so no more CPU->GPU bottleneck like in bullet trails. 
- Players display dance animation (now it's just walking before Joanne finishes all 4 dance animations)
- Players dancing cannot move with WASD, but can look around and jump. 

End dance:
- the dance bomb turns back into an egg
- egg lerps towards warren bear (which can be any random location tho, to avoid farming)

Testing only:
- Client 0 press WASD together would activate the dancebomb. 
- whoever sits at seat 0 during the demo will have to activate it lol. 

Untitled

----
~~nice2haves not in there yet~~
- meme song during dance

### CODE REVIEW

**Comment by Alanine42** [comment]
new changes

- change song during dance

- merged from main

**Review by AnnLe4869 — APPROVED** [review]
Okay let's go
